### PR TITLE
refactor: reusable thread pool to decrypt keystores

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

Decrypting keystores is required in multiple places
- when importing from keystore definition files ([#5357](https://github.com/ChainSafe/lodestar/pull/5357))
- when importing keystores through the keymanager API ([#5438](https://github.com/ChainSafe/lodestar/issues/5438))

This PR makes the thread pool implementation reusable, improves pool lifecycle management and error handling.

**Description**

Reusable thread pool to decrypt keystores, addresses the following issues
- terminates workers if process receives exit signal
- ensures consistent configuration of thread pool
- enables worker to decrypt from file or keystore string
- more graceful and deterministic error handling
- improved error handling and formatting 👇 

**Current error handling**

Current error thrown if decryption of keystores fails is not that helpful if there are multiple. There is no indication on which keystore file is affected and only reports one error.

```
✖ Error: Invalid password
    at Object.<anonymous> (/home/devops/goerli/lodestar/node_modules/@chainsafe/bls-keystore/lib/functional.js:86:19)
    at Generator.next (<anonymous>)
    at fulfilled (/home/devops/goerli/lodestar/node_modules/@chainsafe/bls-keystore/lib/functional.js:5:58)
```

**Updated error handling**

Updated error handling provides more details on which keystore import failed and why it failed.

Single keystore import error

```
 ✖ Error importing keystore

keystore-m_12381_3600_2_0_0-1673696854.json: Invalid password
```

Multiple keystore import errors

```
 ✖ Multiple errors importing keystores

keystore-m_12381_3600_2_0_0-1673696854.json: Invalid password
keystore-m_12381_3600_8_0_0-1673696857.json: Invalid password
keystore-m_12381_3600_5_0_0-1673696855.json: Invalid password
```

Single error, multiple aborted due to error

```
 ✖ Error importing keystore

keystore-malformed.json: Unexpected token s in JSON at position 0

Aborted 100 pending keystore imports
```

Multiple errors, multiple aborted due to errors

```
 ✖ Multiple errors importing keystores

keystore-m_12381_3600_1028_0_0-1680958115.json: Invalid password
keystore-m_12381_3600_1031_0_0-1680958117.json: Invalid password
keystore-m_12381_3600_1032_0_0-1680958118.json: Invalid password
keystore-m_12381_3600_1045_0_0-1680958125.json: Invalid password
keystore-m_12381_3600_1049_0_0-1680958127.json: Invalid password
keystore-m_12381_3600_1059_0_0-1680958133.json: Invalid password
keystore-m_12381_3600_1018_0_0-1680958110.json: Invalid password
keystore-m_12381_3600_1034_0_0-1680958119.json: Invalid password
keystore-m_12381_3600_1087_0_0-1680958149.json: Invalid password

Aborted 91 pending keystore imports
```

### CODE REVIEW

File: packages/cli/src/cmds/validator/handler.ts [hunk 1]
```diff
@@ -74,6 +74,13 @@ export async function validatorHandler(args: IValidatorCliArgs & GlobalArgs): Pr
   // This AbortController interrupts various validators ops: genesis req, clients call, clock etc
   const abortController = new AbortController();
 
+  // We set infinity for abort controller used for validator operations,
+  // to prevent MaxListenersExceededWarning which get logged when listeners > 10
+  // Since it is perfectly fine to have listeners > 10
+  setMaxListeners(Infinity, abortController.signal);
+
+  onGracefulShutdownCbs.push(async () => abortController.abort());
```
Comment (nflaig) [inline packages/cli/src/cmds/validator/handler.ts:82]: Abort has to be pushed to graceful shutdown callbacks before invoking `getSignersFromArgs`.

File: packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts
Comment (nflaig) [inline packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts:1]: Setting `DEBUG=threads:pool:*` is pretty helpful to analyse thread and task lifecycle

File: packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      () =>
+        spawn<DecryptKeystoreWorkerAPI>(new Worker("./worker.js"), {
+          // The number below is big enough to almost disable the timeout
+          // which helps during tests run on unpredictably slow hosts
+          timeout: 5 * 60 * 1000,
+        }),
+      {
+        // Adjust worker pool size based on keystore count
+        size: Math.min(keystoreCount, maxPoolSize),
+        // Decrypt keystores in sequence, increasing concurrency does not improve performance
```
Comment (nflaig) [inline packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts:24]: Did several test runs with different concurrency but it seems to just overload the threads and throttle the CPU

File: packages/cli/src/cmds/validator/signers/importExternalKeystores.ts [hunk 3]
```diff
@@ -37,9 +37,6 @@ export async function readPassphraseOrPrompt(args: {importKeystoresPassword?: st
       },
     ]);
 
-    // eslint-disable-next-line no-console
-    console.log("Password is correct");
```
Comment (nflaig) [inline packages/cli/src/cmds/validator/signers/importExternalKeystores.ts:41]: This console.log is incorrect and not helpful as password verification is done later in the process

**Review by wemeetagain — DISMISSED** [review]
Great job on the error handling.
Also having a cancelable `decryptKeystoreDefinitions` is really nice.

**Comment by nflaig** [comment]
> Also having a cancelable decryptKeystoreDefinitions is really nice.

Yeah, that one is quite important because if you force close the process then the .lock files are not removed and need to be manually cleaned up.

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

**Pull request merged**